arch/arm/src/stm32h7/stm32h7x3xx_rcc.c

```diff
@@ -866,30 +885,25 @@ void stm32_stdclockconfig(void)
         {
         }
 
-      /* Over-drive is needed if
-       *  - Voltage output scale 1 mode is selected and SYSCLK frequency is
-       *    over 400 MHz.
-       */
+#if STM32_VOS_OVERDRIVE && (STM32_PWR_VOS_SCALE == PWR_D3CR_VOS_SCALE_1)
+      /* Over-drive support for VOS1 */
 
-      if ((STM32_PWR_VOS_SCALE == PWR_D3CR_VOS_SCALE_1) &&
-           STM32_SYSCLK_FREQUENCY > 400000000)
-        {
-          /* Enable System configuration controller clock to Enable ODEN */
+      /* Enable System configuration controller clock to Enable ODEN */
```

Comment:
Note that silicon revision Y of STM32H7x3 has an erratum that prevents being able to turn on overdrive, so those microcontrollers are limited to 400 MHz. This is fixed in all subsequent revisions and all new chips do correctly support overdrive.

I am not sure if that is the reason why we had `if ((STM32_PWR_VOS_SCALE == PWR_D3CR_VOS_SCALE_1) && STM32_SYSCLK_FREQUENCY > 400000000)` here.

Reply:
This problem is not related to SYSCLK frequency. I run uc with SYSCLK=400MHz, but somehow overdrive is required for external ULPI even if I don't set the maximum frequency. I submitted a ticket for ST support, maybe they can tell something more about it. But I'm afraid they won't say anything, there are threads on their forum where ST aren't very willing to help in ULPI's matters. The entire STM32 OTGHS IP block (from Synopsys) is a garbage :)